engine/src/test/java/io/zeebe/engine/processor/workflow/subprocess/EventSubprocessTest.java

```diff
@@ -0,0 +1,217 @@
+            .skipUntil(r -> r.getPosition() > jobCompletePosition)
+            .limit(9)
+            .asList();
+
+    assertThat(events)
+        .extracting(Record::getIntent, e -> e.getValue().getElementId())
+        .containsExactly(
+            tuple(WorkflowInstanceIntent.ELEMENT_COMPLETING, "event_sub_task"),
+            tuple(WorkflowInstanceIntent.ELEMENT_COMPLETED, "event_sub_task"),
+            tuple(WorkflowInstanceIntent.SEQUENCE_FLOW_TAKEN, "after_task_flow"),
+            tuple(WorkflowInstanceIntent.ELEMENT_ACTIVATING, "event_sub_end"),
+            tuple(WorkflowInstanceIntent.ELEMENT_ACTIVATED, "event_sub_end"),
+            tuple(WorkflowInstanceIntent.ELEMENT_COMPLETING, "event_sub_end"),
+            tuple(WorkflowInstanceIntent.ELEMENT_COMPLETED, "event_sub_end"),
+            tuple(WorkflowInstanceIntent.ELEMENT_COMPLETING, "event_sub_proc"),
+            tuple(WorkflowInstanceIntent.ELEMENT_COMPLETED, "event_sub_proc"));
+  }
+
+  private void assertEventSubprocessLifecycle(Record<WorkflowInstanceRecordValue> timerTriggered) {
+    final List<Record<WorkflowInstanceRecordValue>> events =
+        RecordingExporter.workflowInstanceRecords()
+            .skipUntil(r -> r.getPosition() > timerTriggered.getPosition())
+            .limit(13)
+            .asList();
+
+    assertThat(events)
+        .extracting(Record::getIntent, e -> e.getValue().getElementId())
+        .containsExactly(
+            tuple(WorkflowInstanceIntent.ELEMENT_ACTIVATING, "event_sub_proc"),
+            tuple(WorkflowInstanceIntent.ELEMENT_ACTIVATED, "event_sub_proc"),
+            tuple(WorkflowInstanceIntent.ELEMENT_ACTIVATING, "event_sub_start"),
+            tuple(WorkflowInstanceIntent.ELEMENT_ACTIVATED, "event_sub_start"),
+            tuple(WorkflowInstanceIntent.ELEMENT_COMPLETING, "event_sub_start"),
+            tuple(WorkflowInstanceIntent.ELEMENT_COMPLETED, "event_sub_start"),
+            tuple(WorkflowInstanceIntent.SEQUENCE_FLOW_TAKEN, "event_sub_flow"),
+            tuple(WorkflowInstanceIntent.ELEMENT_ACTIVATING, "event_sub_end"),
+            tuple(WorkflowInstanceIntent.ELEMENT_ACTIVATED, "event_sub_end"),
+            tuple(WorkflowInstanceIntent.ELEMENT_COMPLETING, "event_sub_end"),
+            tuple(WorkflowInstanceIntent.ELEMENT_COMPLETED, "event_sub_end"),
+            tuple(WorkflowInstanceIntent.ELEMENT_COMPLETING, "event_sub_proc"),
+            tuple(WorkflowInstanceIntent.ELEMENT_COMPLETED, "event_sub_proc"));
+  }
+
+  private static BpmnModelInstance createEventSubProc() {
+    final ProcessBuilder builder = Bpmn.createExecutableProcess("proc");
+    builder
+        .eventSubProcess("firstSubProcess")
+        .startEvent("start_sub_1")
+        .message(b -> b.name("msg").zeebeCorrelationKey("innerKey"))
+        .endEvent();
+    builder
+        .eventSubProcess("event_sub_proc")
+        .startEvent("event_sub_start")
+        .timerWithDuration("PT1S")
```

Comment:
```suggestion
        .timerWithDuration("PT0S")
```

Minimize the execution time of the test.